unit_tests/log_transfer/test_main_chkpt_info.cpp

```diff
@@ -0,0 +1,872 @@
+			       LOG_DATA_ADDR *addr, int ulength, const char *udata, int rlength, const char *rdata)
+{
+  assert (false);
+}
+
+LOG_LSA
+prior_lsa_next_record (THREAD_ENTRY *thread_p, LOG_PRIOR_NODE *node, log_tdes *tdes)
+{
+  assert (false);
+}
+
+void
+_er_log_debug (const char *file_name, const int line_no, const char *fmt, ...)
+{
+  assert (false);
+}
+
+int
+logtb_reflect_global_unique_stats_to_btree (THREAD_ENTRY *thread_p)
+{
+  assert (false);
+}
+
+int
+pgbuf_flush_checkpoint (THREAD_ENTRY *thread_p, const LOG_LSA *flush_upto_lsa, const LOG_LSA *prev_chkpt_redo_lsa,
+			LOG_LSA *smallest_lsa, int *flushed_page_cnt)
+{
+  assert (false);
+}
+
+int
+fileio_synchronize_all (THREAD_ENTRY *thread_p, bool include_log)
+{
+  assert (false);
+}
+
+int
+csect_exit (THREAD_ENTRY *thread_p, int cs_index)
+{
+  assert (true);
+}
+
+void
+logpb_flush_pages_direct (THREAD_ENTRY *thread_p)
+{
+  assert (false);
+}
+
+LOG_TDES *
+logtb_get_system_tdes (THREAD_ENTRY *thread_p)
+{
+  assert (false);
+}
+
+log_global::log_global ()
+  : m_prior_recver (prior_info)
+{
+}
+
+log_global::~log_global ()
+{
+}
+
+namespace cublog
+{
+  EXPAND_PACKABLE_OBJECT_EMPTY_DEF (meta);
+
+  prior_recver::prior_recver (log_prior_lsa_info &prior_lsa_info)
+    : m_prior_lsa_info (prior_lsa_info)
+  {
+  }
+  prior_recver::~prior_recver () = default;
+}
+
+mvcc_active_tran::mvcc_active_tran () = default;
+mvcc_active_tran::~mvcc_active_tran () = default;
+mvcc_trans_status::mvcc_trans_status () = default;
+mvcc_trans_status::~mvcc_trans_status () = default;
+mvcctable::mvcctable () = default;
+mvcctable::~mvcctable () = default;
+
+log_append_info::log_append_info () = default;
+log_prior_lsa_info::log_prior_lsa_info () = default;
+
+int
+csect_enter (THREAD_ENTRY *thread_p, int cs_index, int wait_secs)
+{
+  assert (true);
+}
+
+void
+log_system_tdes::map_all_tdes (const map_func &func)
+{
+  std::lock_guard<std::mutex> lg (systb_Mutex);
+  for (auto &el : systb_System_tdes)
+    {
+      log_tdes *tdes = el.second;
+      assert (tdes != NULL);
+      func (*tdes);
+    }
+}
+#define NUM_TOTAL_TRAN_INDICES log_Gl.trantable.num_total_indices
+
+void
+logtb_clear_tdes (THREAD_ENTRY *thread_p, LOG_TDES *tdes)
+{
+}
+
+log_tdes *
+systdes_create_tdes ()
+{
+  log_tdes *tdes = new log_tdes ();
+  return tdes;
+}
+
+log_tdes *
+log_system_tdes::rv_get_or_alloc_tdes (TRANID trid)
+{
+  log_tdes *tdes = rv_get_tdes (trid);
+  if (tdes == NULL)
+    {
+      log_tdes *tdes = systdes_create_tdes ();
+      tdes->state = TRAN_UNACTIVE_UNILATERALLY_ABORTED;
+      tdes->trid = trid;
+      systb_System_tdes[trid] = tdes;
+      return tdes;
+    }
+  else
+    {
+      assert (tdes->trid == trid);
+      return tdes;
+    }
+}
+
+log_tdes *
+log_system_tdes::rv_get_tdes (TRANID trid)
+{
+  auto it = systb_System_tdes.find (trid);
+  if (it != systb_System_tdes.end ())
+    {
+      return it->second;
+    }
+  else
+    {
+      return NULL;
+    }
+}
```

Comment:
I think these should be removed too.
```suggestion
log_tdes *
systdes_create_tdes ()
{
  log_tdes *tdes = new log_tdes ();
  return tdes;
}

log_tdes *
log_system_tdes::rv_get_or_alloc_tdes (TRANID trid)
{
  log_tdes *tdes = rv_get_tdes (trid);
  if (tdes == NULL)
    {
      log_tdes *tdes = systdes_create_tdes ();
      tdes->state = TRAN_UNACTIVE_UNILATERALLY_ABORTED;
      tdes->trid = trid;
      systb_System_tdes[trid] = tdes;
      return tdes;
    }
  else
    {
      assert (tdes->trid == trid);
      return tdes;
    }
}

log_tdes *
log_system_tdes::rv_get_tdes (TRANID trid)
{
  auto it = systb_System_tdes.find (trid);
  if (it != systb_System_tdes.end ())
    {
      return it->second;
    }
  else
    {
      return NULL;
    }
}
```